Student Grades

Question 1:
Consider the following midterm grades for students taking math, science, reading, and history
 (each grade is calculated out of a maximum of 20 points). 



In [3]:
import numpy as np
import pandas as pd
import math

In [ ]:
math = [11, 12, 11, 7, 9.5, 15, 18]
science = [9, 11, 18, 15.5, 12, 18.5, 15.5]
reading= [16, 8.5, 11.5, 11, 10.5, 7, 11]
history= [13.5, 10, 9, 14, 14, 12, 7.5]


1. Calculate the mean for each individual subject.

In [17]:
np.mean(math)

11.928571428571429

In [18]:
np.mean(science)

14.214285714285714

In [19]:
np.mean(reading)

10.785714285714286

In [20]:
np.mean(history)

11.428571428571429

2. Find the median grade in Math for all students.

In [21]:
np.median(math)

11.0

3. Calculate the mode for History.

In [16]:
import statistics
statistics.mode(history)
# np.mode(history)

14

Question 2:
Given the following grades from Question 1, calculate the correlation between the different subjects. What two subjects have the strongest correlation


In [22]:
np.corrcoef(math, science)[0,1]

0.2694940962141907

In [23]:
np.corrcoef(math, reading)[0,1]

-0.26856188905797385

In [24]:
np.corrcoef(math, history)[0,1]

-0.7183583464149538

In [25]:
np.corrcoef(science, reading)[0,1]

-0.5276058297032529

In [26]:
np.corrcoef(science, history)[0,1]

-0.35868815565295303

In [27]:
np.corrcoef(reading, history)[0,1]

0.20679404264832346

Math and science have the strongest correlation compared to the other subjects

Question 3:
Consider the following numbers representing the grades from the most recent exam in a math clas


In [42]:
def desc_stats(grades):
    mean = round(np.mean(grades), 2)
    print('Mean: ', mean)
    median = np.median(grades)
    print('Median: ', median)
    mode = statistics.mode(grades)
    print('Mode: ', mode)
    mini = np.min(grades)
    print('Minimum: ', mini)
    maxi = np.max(grades)
    print('Maximum: ', maxi)
    range = np.ptp(grades)
    print('Range: ', range)
    variance = round(np.var(grades), 2)
    print('Variance: ', variance)
    std_dev = round(np.std(grades), 2)
    print('Standard Deviation: ', std_dev)


In [43]:
grades = [24, 5, 15, 60, 54, 82, 99, 80, 70, 98, 93, 60, 33, 22, 65, 61, 51, 58, 83, 86, 42, 67, 60]
desc_stats(grades)

Mean:  59.48
Median:  60.0
Mode:  60
Minimum:  5
Maximum:  99
Range:  94
Variance:  664.16
Standard Deviation:  25.77


Extended Challenge:


Interpret the results. If you were the instructor of the class, would you curve the grade?


 Assume:
0-100 is the possible range of scores
The grade scale is:
F: 0-60
D: 60-70
C: 70-80
B: 80-90
A
: 90-100
Think about the grade scale and use the information from your desc_stats() to justify your results. 

Answer:

The class has a mean of 59.48  which is equivalent to F in grade scale. The median is 60 which is equivalent to D in grade scale.
We can interpret that the class has shown poor performance in terms of grade scale.